src/rgw/rgw_rest.cc

```diff
@@ -1486,70 +1486,59 @@ static int read_all_chunked_input(req_state *s, char **pdata, int *plen, const u
 	need_to_read *= 2;
 
       if ((unsigned)total > max_read) {
-	free(data);
-	return -ERANGE;
+	return std::make_tuple(-ERANGE, std::move(pdata), 0);
       }
       total += need_to_read;
-
-      void *p = realloc(data, total + 1);
-      if (!p) {
-	free(data);
-	return -ENOMEM;
-      }
-      data = (char *)p;
```

Comment:
i think there needs to be some logic to grow the vector here, ie data.resize(len+1)?

Reply:
Yeah ok, so vector will resize automatically only when some methods are invoked, not when indexing the vector. Thanks for pointing that out, will fix that.